### DISCUSSION TITLE
Feature request: Sync cooking of recipe when it's been planned in the meal plan

### DISCUSSION THREAD

**Discussion by sortab** [TandoorRecipes/recipes#1659] [Ideas]
In relation to my previous question I'd like to suggest a feature:

As the site works now one has to both add a recipe to the meal plan AND log the recipe has been cooked at a specific date  in order to use the option to sort by "last cooked" / "no. times cooked" when searching recipes. A synchronization between the meal planner and of a recipe being logged as cooked would be great.

My previous question:
https://github.com/TandoorRecipes/recipes/discussions/1650#discussioncomment-2395408

**Comment by sortab** [comment]
If it was possible to sort recipes by "last planned" and "number of times planned" that would also be a solution to the use case I have of this excellent software.

**Comment by vabene1111** [comment]
hmm interesting, i tought about this before as well. Basically planning and logging are two sides of the same things. But sometimes you plan and then dont follow that plan so there needs to be room for the exception. On the other hand it does not make sense to implement all those features twice (e.g. the search filters for last cooked/last planned). i guess somehow merging these two things makes sense. 

One question there however would be how to handle moving a plan entry afterwards. The log would need to be connected to the plan or logging would just add a plan item that is shown/not shown ? maybe the plan view could have a filter to "show log only entries" (merging CookLog and PlanEntry models into one)

**Comment by sortab** [comment]
In my (and my wife's) opinion , it would be sufficient just to use the planning aspect (without the need for logging also), because if we changed our minds or the plan changed IRL, we would update the meal plan to reflect that change.

Merging the cooking log and the plan entry does sound like a possible solution. If it's too much of a change, perhaps a specific sync feature between the two would suffice. And if there was a setting where you could change the default behavior so it either synced or did not sync by default. 
This would add even more flexibility to the system, and perhaps other users have different use cases that are different than mine/ours which would not benefit from it always being synced (or merged for that matter).

**Reply by smilerz** [reply]
To keep things as clean as possible, making 'last cooked' a defined preference between using cooklog or mealplan or just having both filters available makes the most sense.  having mealplans automatically log would be complicated and likely lead to further defects.

One alternative would be a view that asks: Log these Mealplans that would allow rating/logging in bulk.

**Reply by sortab** [reply]
Defining a preference for 'last cooked' (cooklog or mealplan) would also be a great feature.
